Jenkinsfile

```diff
@@ -1,59 +1,100 @@
-node('CAE-Jenkins2-DH-Agents-Linux'){
-    def mdk_builder
-
-    stage('Preparation') {
-        checkout scm
+pipeline {
+    agent {
+        docker{
+            image 'circleci/openjdk:8'
+        }
+    }
+    environment {
+        GRADLE_USER_HOME = "$WORKSPACE/.gradle"
+        ARTIFACTORY_CREDENTIALS = credentials('mdk-artifactory-credentials')
+        TESTRAIL_CREDENTIALS = credentials('mdk-testrail-credentials')
+        BUILD_ACCESS = credentials('mdk-build-access')
+        TESTRAIL_HOST = credentials('mdk-testrail-host')
+        TESTRAIL_SUITE_ID = credentials('mdk-testrail-suite-id')
+        ARTIFACTORY_URL = credentials('mdk-artifactory-url')
+        ADDITIONAL_TEST_ARGUMENTS = credentials('mdk-additional-test-args-19')
     }
 
-    sh 'git branch --list'
-    GIT_TAG = sh (
-                script: '$(git describe --tags --exact-match `git rev-parse HEAD 2> /dev/null` 2> /dev/null) || true',
-                returnStdout: true
-                ).trim()
-    echo "GIT TAG = $GIT_TAG"
-
-    mdk_builder = docker.build("willard/mdk-builder", "-f Dockerfile .")
-
-withCredentials([
-    usernamePassword(credentialsId: 'mdk-testrail-credentials',      usernameVariable: 'TESTRAIL_CREDENTIALS_USR',    passwordVariable: 'TESTRAIL_CREDENTIALS_PSW' ),
-    usernamePassword(credentialsId: 'mdk-artifactory-credentials',   usernameVariable: 'ARTIFACTORY_CREDENTIALS_USR', passwordVariable: 'ARTIFACTORY_CREDENTIALS_PSW' ),
-    string(          credentialsId: 'mdk-build-access',              variable: 'BUILD_ACCESS'),
-    string(          credentialsId: 'mdk-testrail-host',             variable: 'TESTRAIL_HOST'),
-    string(          credentialsId: 'mdk-testrail-suite-id',         variable: 'TESTRAIL_SUITE_ID'),
-    string(          credentialsId: 'mdk-artifactory-url',           variable: 'ARTIFACTORY_URL'),
-    string(          credentialsId: 'mdk-additional-test-args-19',   variable: 'ADDITIONAL_TEST_ARGUMENTS')]) {
-
+    stages {
         stage('Dependencies') {
-            mdk_builder.inside{
-                sh '''
-                export GRADLE_USER_HOME=$(pwd)/.gradle
-                ./gradlew -PbuildNumber=$BUILD_NUMBER -PbuildAccess=$BUILD_ACCESS -PbuildTag=$GIT_TAG -PartifactoryUrl=$ARTIFACTORY_URL -PartifactoryUsername=$ARTIFACTORY_CREDENTIALS_USR -PartifactoryPassword=$ARTIFACTORY_CREDENTIALS_PSW dependencies --gradle-user-home GRADLE_USER_HOME --info --stacktrace --refresh-dependencies
-                '''
+            steps {
+                echo 'TAG_NAME = $TAG_NAME'
+                sh './gradlew \
+                    -PbuildNumber=$BUILD_NUMBER \
+                    -PbuildAccess=$BUILD_ACCESS \
+                    -PbuildTag=$TAG_NAME \
```

Comment:
Where is $TAG_NAME defined?

Reply:
The Jenkins branch api plugin sets it when a multibranch pipeline is set to discover tags.